packages/terra-table/src/SelectableTableRows.jsx

```diff
@@ -31,24 +35,77 @@ const propTypes = {
 const defaultProps = {
   disableUnselectedRows: false,
   onChange: undefined,
+  onBlur: undefined,
   selectedIndexes: [],
 };
 
+const rowsSelector = 'row-group';
+
 class SelectableTableRows extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+
+  componentDidMount() {
+    // Adds ARIA-Label for Pre-selected rows
+    const rowGroup = document.querySelector(`#${rowsSelector}`);
+    for (let rowIndex = 0; rowIndex < rowGroup.children.length; rowIndex += 1) {
+      const preselectedRow = rowGroup.children[rowIndex];
+      let ariaLabel = 'pre-selected row';
+      if ((preselectedRow.hasAttribute('aria-selected')) && (preselectedRow.getAttributeNode('aria-selected').value === 'true')) {
+        for (let columnIndex = 0; columnIndex < preselectedRow.children.length; columnIndex += 1) {
+          ariaLabel += ` ${preselectedRow.children[columnIndex].innerText}`;
+        }
+        rowGroup.children[rowIndex].setAttribute('aria-label', ariaLabel);
+      }
+    }
   }
 
   handleOnChange(event, index) {
     if (this.props.onChange) {
       this.props.onChange(event, index);
     }
+
+    if (!document.activeElement.hasAttribute('aria-label')) {
+      // Aria-Label is being populated with the cell contents for screen reader to read it as whole instead of cell value.
+      const ariaLabel = 'row selected';
+      document.activeElement.setAttribute('aria-label', ariaLabel);
+    }
+  }
+
+  handleOnBlur(event, index) {
+    if (this.props.onBlur) {
+      this.props.onBlur(event, index);
+    }
+  }
+
+  wrappedOnBlur(row, index) {
+    const initialOnBlur = row.props.onBlur;
+    // removes aria-label from unselected row
+    return (event) => {
+      if (row.props.isSelectable !== false) {
+        this.handleOnBlur(event, index);
+      }
+
+      const rowGroup = document.querySelector(`#${rowsSelector}`);
+      const previousRow = rowGroup.children[index];
```

Comment:
You can rename the variable as selectedRow and rowGroup as SelectedRowGroup.

Reply:
[updated to selected row](https://github.com/cerner/terra-core/blob/153a90093ca106ce109d4c4d0ab861ec611985b2/packages/terra-table/src/SelectableTableRows.jsx#L77) 👍